js/ui/animation.js

```diff
@@ -54,25 +54,37 @@ var Animation = new Lang.Class({
         this._isPlaying = false;
     },
 
+    _getCurrentFrame: function() {
+        return this._framesInfo[this._frameIndex];
+    },
+
+    _getCurrentFrameActor: function() {
+        let currentFrame = this._getCurrentFrame();
+        return this._animations.get_child_at_index(currentFrame.frameIndex);
```

Comment:
Sorry I only caught this now, but if you call the other field `delay`, maybe this one should be simply called `index`? (Or the other one called `frameDelay`, just looking for consistency)